funfact/lang/_tsrex.py

```diff
@@ -279,41 +282,59 @@ class TsrEx(_BaseEx, ArithmeticMixin, IndexRenamingMixin, TranspositionMixin):
 
     For more details, see the corresponding
     [user guide page](../../pages/user-guide/tsrex).
-
     '''
-    pass
 
 
-class IndexEx(_BaseEx):
-    def __invert__(self):
-        '''Implements the `~i` syntax.'''
-        return IndexEx(dataclasses.replace(self.root, bound=True, kron=False))
+_dispatch = Dispatcher()
```

Comment:
I don't fully understand the need for multiple dispatch here. There seems to be only one implementation of every function? Is it to raise an error if it is called with incorrect input?

Reply:
You are right :smile: There are two purposes:
1. We will need a redirection anyway to handle `__op__` and `__rop__` in a single place.
2. Using multiple dispatch is a systematic way to catch certain types of syntax errors.